src/concepts/metric_hub.md

```diff
@@ -82,6 +82,27 @@ from_expression = """
 submission_date_column = "submission_date"
 ```
 
+Data sources can be joined with other data sources:
+
+```toml
+# Join the `baseline` data source with the `metrics` data source.
+# Definitions for both data sources must exist.
+[data_sources.baseline.joins.metrics]
+relationship = "many_to_many"  # this determines the type of JOIN used; options: many_to_many, one_to_one, one_to_many, many_to_one
+on_expression = """  # SQL expression specifying the JOIN condition
+  baseline.client_id = metrics.client_id AND
+  baseline.submission_date = metrics.submission_date
```

Comment:
We should make sure that the default behavior is explained somewhere for each of these fields (i.e., how does the join work if they are not defined).

Reply:
I added some more context and a clearer example of the default values to the `join` expression below this one